### PR TITLE
Wasmvm 1.4 upgrade

### LINKED ISSUE #1586: Upgrade to wasmvm 1.4
I released wasmvm v1.4.0-rc.1 today.

There are some new distribution queries that need to be implemented: [DistributionQuery::{DelegationRewards, DelegationTotalRewards, DelegatorValidators}](https://github.com/CosmWasm/wasmvm/blob/7c2a35d6d96d398e4ec4c82f6f68effbbd2873cd/types/queries.go#L369-L374)

### PR DESCRIPTION
**Pull request by alpe**
Resolves #1586 

Includes 
* Wasmvm 1.4 upgrade
* Dockerfile
* Capabilty
* Distribution queries

### CODE REVIEW

File: x/wasm/keeper/testdata/contracts.go [hunk 1]
```diff
@@ -7,6 +7,10 @@ import (
 	"github.com/cosmos/cosmos-sdk/types"
 )
 
+const (
+	ChecksumHackatom = "EBED57113226028E105A28B02C473520C7D406A0E55AE85078CB3F036299C1E4"
```
Comment (webmaster128) [inline x/wasm/keeper/testdata/contracts.go:11]: Could we keep this lower case? The checksums of CosmWasm contracts are all lower case hex throughout the stack

File: x/wasm/keeper/testdata/version.txt [hunk 2]
```diff
@@ -1,3 +1,2 @@
-v1.3.0
+v1.4.0-rc.1
 burner.wasm: v1.2.0
```
Comment (webmaster128) [inline x/wasm/keeper/testdata/version.txt:2]: You can update burner.wasm too now (https://github.com/CosmWasm/cosmwasm/issues/1783#issuecomment-1697302795)

**Review by webmaster128 — COMMENTED** [review]
Some first thoughts

File: x/wasm/keeper/testdata/version.txt (hunk 2 above)
Comment (alpe) [reply x/wasm/keeper/testdata/version.txt:2]: I have tried with burner v1.4.0 but not all data is erased. https://github.com/CosmWasm/wasmd/blob/v0.41.0/x/wasm/keeper/keeper_test.go#L1418-L1419 fails

Comment (webmaster128) [reply x/wasm/keeper/testdata/version.txt:2]: Did you apply the limit as shown in the linked diff? Maybe it needs to be higher. You can put 100 or 1000 there.

Yeah the "action" attribute has changed to "migrate".

**Review by webmaster128 — APPROVED** [review]
LGTM

File: x/wasm/keeper/keeper_test.go [hunk 3]
```diff
@@ -42,7 +43,7 @@ import (
 //go:embed testdata/hackatom.wasm
 var hackatomWasm []byte
 
-const AvailableCapabilities = "iterator,staking,stargate,cosmwasm_1_1"
+const AvailableCapabilities = "iterator,staking,stargate,cosmwasm_1_1,cosmwasm_1_2,cosmwasm_1_3,cosmwasm_1_4"
```
Comment (pinosu) [inline x/wasm/keeper/keeper_test.go:46]: 👍

File: x/wasm/keeper/query_plugin_integration_test.go [hunk 4]
```diff
@@ -0,0 +1,694 @@
...
+				assert.Contains(t, gotErr.Error(), fmt.Sprintf("codespace: %s, code: %d:", spec.expErr.Codespace(), spec.expErr.ABCICode()))
+				return
+			}
+			require.NoError(t, gotErr)
+			assert.JSONEq(t, string(spec.exp), string(gotData), string(gotData))
+		})
+	}
+}
+
+func TestDistributionQuery(t *testing.T) {
```
Comment (pinosu) [inline x/wasm/keeper/query_plugin_integration_test.go:476]: very good tests!

File: x/wasm/keeper/query_plugins_test.go [hunk 5]
```diff
@@ -763,6 +764,7 @@ func TestAcceptListStargateQuerier(t *testing.T) {
 }
 
 func TestDistributionQuerier(t *testing.T) {
```
Comment (pinosu) [inline x/wasm/keeper/query_plugins_test.go:766]: Do you think this test can be deleted?

**Review by pinosu — APPROVED** [review]
Nice work! LGTM 👍

File: x/wasm/keeper/query_plugins_test.go (hunk 5 above)
Comment (alpe) [reply x/wasm/keeper/query_plugins_test.go:766]: ja, I will remove the test. Thanks for finding this!

**Pull request merged**